eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -2282,4 +2347,67 @@ private function deleteTranslationFromContentVersions($contentId, $languageId, $
             );
         }
     }
+
+    /**
+     * Get query builder for content version objects, used for version loading w/o fields.
+     *
+     * Creates a select query with all necessary joins to fetch a complete
+     * content object. Does not apply any WHERE conditions, and does not contain
+     * name data as it will lead to large result set {@see createNamesQuery}.
+     *
+     * @return \Doctrine\DBAL\Query\QueryBuilder
+     */
+    private function createVersionInfoFindQueryBuilder(): DoctrineQueryBuilder
```

Comment:
You can _(by now)_ instead reuse `->queryBuilder-> createVersionInfoQueryBuilder()` which was just added as part of #2739

But I see you improved it by setting `ezcontentobject_version` as from, so feel free to adapt it to your approach (while retaining possibility to specify version number) if that makes sense here.

Reply:
EDIT: Partly ignore this, unless we refactor it to work for your use case it won't atm _(it expects to give current version or a provided version number only atm, and this one instead uses version table in from which is better in this case)_